packages/uows-client-generator/src/index.ts

```diff
@@ -86,60 +82,92 @@ const makeArgsObjTemplate: string =
-                functArg[1] = functDesc.input[element]
-                if (element.includes("[]")) functArg[1] += "[]";
-            }
-
-            functArgs[index] = functArg;
-        }
-    });
-    return functArgs;
+  const functArgs: string[][] = [];
+  Object.keys(functDesc.input).forEach((element: string, index: number) => {
+    if (element !== 'targetNSAlias' && element !== 'targetNamespace') {
+      const functArg: string[] = [];
+      functArg[0] = element.replace('[]', '');
+
+      if (typeof functDesc.input[element] === 'object') {
+        functArg[1] = element; //if the argument is a complex type, assume the name of the argument is the same as the name of it's type
+      } else {
+        functArg[1] = functDesc.input[element];
+        if (element.includes('[]')) functArg[1] += '[]';
+      }
+
+      functArgs[index] = functArg;
+    }
+  });
+
+  return functArgs;
+};
+
+//Creates a wrapper function for each function exposed by a webservice and returns them in an object
+const populateFuncts = (): any => {
+  const functObj: any = {};
+  const serviceDesc: any = wsdlDesc[Object.keys(wsdlDesc)[0]];
+  const collectionOfFunctions: any = serviceDesc[Object.keys(serviceDesc)[0]];
+  Object.keys(collectionOfFunctions).forEach((element: string) => {
+    const functArgDetails: string[][] = createFunctArgDetails(
+      collectionOfFunctions[element]
+    );
+    functObj[element] = createFunctTemplate(element, functArgDetails);
+  });
+
+  return functObj;
 };
 
 //Writes the UOWSSoapInterface.ts file to storage
 const generateCode = (obj: any): void => {
-    fs.writeFileSync(filePath, `import * as soap from "soap";\n\n`);
-    fs.appendFileSync(filePath, `export default class UOWSSoapClient {\n\n`);
-    fs.appendFileSync(filePath, `private wsdlUrl: string;\n`);
-    fs.appendFileSync(filePath, `private wsdlDesc: any = ${JSON.stringify(wsdlDesc)};\n\n`);
-    fs.appendFileSync(filePath, constructorTemplate);
-
-    Object.keys(obj).forEach((element: string) => {
-        fs.appendFileSync(filePath, obj[element]);
-    });
+  let filePath: string;
+
+  if (process.argv[3]) filePath = process.argv[3];
+  else filePath = './src/UOWSSoapInterface.ts';
+
+  // eslint-disable-next-line prettier/prettier
+  fs.writeFileSync(filePath, 'import * as soap from \'soap\';\n\n');
+  fs.appendFileSync(filePath, 'export default class UOWSSoapClient {\n\n');
+  fs.appendFileSync(filePath, 'private wsdlUrl: string;\n');
+  fs.appendFileSync(
+    filePath,
+    `private wsdlDesc: any = ${JSON.stringify(wsdlDesc)};\n\n`
+  );
+  fs.appendFileSync(filePath, constructorTemplate);
+
+  Object.keys(obj).forEach((element: string) => {
+    fs.appendFileSync(filePath, obj[element]);
+  });
+
+  fs.appendFileSync(filePath, `${makeArgsObjTemplate}\n}`);
+};
 
-    fs.appendFileSync(filePath, `${makeArgsObjTemplate}\n}`);
+//Creates the UOWSService.ts file
+const createInterface = (): void => {
+  if (process.argv[2]) defaultWsdl = process.argv[2];
```

Comment:
Very minor, but could we go back to having `defaultWsdl` as a `const` and assign the argument to a new variable? Something like:
```js
const wsdlUrl = (process.argv[2]) ? process.argv[2] : defaultWsdl;
soap.createClientAsync(wsdlUrl);
```

It won't make a real difference in this case because the program exits after executing `createInterface()`, but if we assign `defaultWsdl` to the last URL supplied by the user then that value would become the default for all other runs too.